google/resource_cloudiot_registry_test.go

```diff
@@ -177,3 +262,49 @@ resource "google_cloudiot_registry" "foobar" {
 }
 `, acctest.RandString(10), acctest.RandString(10), registryName)
 }
+
+func testAccCloudIoTRegistry_singleEventNotificationConfig(topic, registryName string) string {
+	return fmt.Sprintf(`
+resource "google_project_iam_binding" "cloud-iot-iam-binding" {
+  members = ["serviceAccount:cloud-iot@system.gserviceaccount.com"]
+  role    = "roles/pubsub.publisher"
+}
+
+resource "google_pubsub_topic" "event-topic" {
+  name = "%s"
+}
+
+resource "google_cloudiot_registry" "foobar" {
+  depends_on = ["google_project_iam_binding.cloud-iot-iam-binding"]
+
+  name = "%s"
+
+  event_notification_config = {
+    pubsub_topic_name = "${google_pubsub_topic.event-topic.id}"
+  }
+}
+`, topic, registryName)
+}
+
+func testAccCloudIoTRegistry_pluralEventNotificationConfigs(topic, registryName string) string {
+	return fmt.Sprintf(`
+resource "google_project_iam_binding" "cloud-iot-iam-binding" {
+  members = ["serviceAccount:cloud-iot@system.gserviceaccount.com"]
+  role    = "roles/pubsub.publisher"
+}
+
+resource "google_pubsub_topic" "event-topic" {
+  name = "%s"
+}
+
+resource "google_cloudiot_registry" "foobar" {
+  depends_on = ["google_project_iam_binding.cloud-iot-iam-binding"]
+
+  name = "%s"
+
+  event_notification_config = {
```

Comment:
Shouldn't have this been made plural?

Reply:
Moved discussion to upstream PR => https://github.com/GoogleCloudPlatform/magic-modules/pull/2206/files#r320172420